app/models/user.rb

```diff
@@ -1,9 +1,11 @@
 class User < ApplicationRecord
-  has_many :posts
-  has_many :comments
-  has_many :likes
+  has_many :posts, foreign_key: 'author_id', class_name: 'Post'
+  has_many :comments, foreign_key: 'author_id', class_name: 'Comment'
+  has_many :likes, foreign_key: 'author_id', class_name: 'Like'
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   def return_recent_posts
-    Post.where(users_id: id).order(created_at: :DESC).limit(3)
+    Post.where(author_id: id).order(created_at: :DESC).limit(3)
```

Comment:
- [ ]  You should make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
N+1 problem still persists 

Tip: You can use the bullet gem to detect N+1 Problems